### ISSUE TITLE
Re-use this model for student project

### ISSUE THREAD

**Issue by testpilot0** [lightingghost/chemopt#6]
I would be curious to re-use this code to try optimize simple Alcohol Mixtures synthesis. I might have 4 or 5 parameters to control alcohol synthesis reaction. could you advice which part of code I should adjust?

Thanks in advance for any comments.

**Comment by lightingghost** [comment]
You should change the “num_param” to 4 or 5. Also, you can play with different hyper-parameters like hidden_size, num_layers, num_steps, unroll_length, policy, etc.

**Comment by cadmoska** [comment]
Same here ! As I understood from your paper, you encourage a pretraining phase on similar and even dissimilar reactions, since the function that describes the reaction is within the family of gaussian mixtures. Does this means that the the multivariate function that you are trying to optimize is unknown ? Do we need to change the reward function and/or the environment when we want to apply the algorithm to a new synthesis reaction or is it enough to change "num_param", adapt the hyper parameters, provide experimental results then we're good to go ? Thank you in advance @lightingghost @testpilot0

**Comment by lightingghost** [comment]
> Same here ! As I understood from your paper, you encourage a pretraining phase on similar and even dissimilar reactions, since the function that describes the reaction is within the family of gaussian mixtures. Does this means that the the multivariate function that you are trying to optimize is unknown ? Do we need to change the reward function and/or the environment when we want to apply the algorithm to a new synthesis reaction or is it enough to change "num_param", adapt the hyper parameters, provide experimental results then we're good to go ? Thank you in advance @lightingghost @testpilot0

You may want to change the num_param to the number you want and tune other hyperparameters. Please note that optimizing a high-dimensional function is a significantly harder task, so you may need more effort to tune the hyperparams.

**Comment by cadmoska** [comment]
The idea behind my question is that I have an idea of the system behind the reaction I want to optimize, but I wrote it purely based on heuristics. Should I adapt the reward function and the environment or is it not required ? Thank you so much for your first answer

**Comment by lightingghost** [comment]
> The idea behind my question is that I have an idea of the system behind the reaction I want to optimize, but I wrote it purely based on heuristics. Should I adapt the reward function and the environment or is it not required ? Thank you so much for your first answer

If you mean the objective function (mapping from the reaction parameters to the reaction outcome) by "reward function", the exact reward is not required. As the mixture of gaussian should be able to approximate any function.
